vllm/entrypoints/openai/serving_chat.py

```diff
@@ -180,15 +254,27 @@ async def create_chat_completion(
         except ValueError as e:
             return self.create_error_response(str(e))
 
-        result_generator = self.engine.generate(
-            {
-                "prompt": prompt_text,
-                "prompt_token_ids": prompt_ids
-            },
-            sampling_params,
-            request_id,
-            lora_request,
-        )
+        if self.engine.engine.vision_language_config is None:
```

Comment:
There is quite a bit of repetition here. I have updated #4200 with a simpler version of this.

Reply:
Hmm...I guess `multi_modal_data` will always be `None` if `vision_language_config` is None, so yes we can simplify this.